File: tests/Generator/resource/python/complex/property_type.py
from pydantic import BaseModel, Field, GetCoreSchemaHandler, Tag
from pydantic_core import CoreSchema, core_schema
from typing import Any, Dict, Generic, List, Optional, TypeVar, Annotated, Union


# Base property type
class PropertyType(BaseModel):
    deprecated: Optional[bool] = Field(default=None, alias="deprecated")
    description: Optional[str] = Field(default=None, alias="description")
    nullable: Optional[bool] = Field(default=None, alias="nullable")
    type: Optional[str] = Field(default=None, alias="type")
    pass

File: tests/Generator/resource/kotlin/complex/PropertyType.kt

import com.fasterxml.jackson.annotation.*

/**
 * Base property type
 */
open abstract class PropertyType {
    @JsonProperty("deprecated") var deprecated: Boolean? = null
    @JsonProperty("description") var description: String? = null
    @JsonProperty("nullable") var nullable: Boolean? = null
    @JsonProperty("type") var type: String? = null
}

